gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/FolderCompletionContributor.java

```diff
@@ -43,11 +43,11 @@ private void extendCompletion(IElementType elementType) {
             protected void addCompletions(
                     CompletionParameters parameters, ProcessingContext context, CompletionResultSet resultSet) {
 
-                List<String> repositories = List.of("https://repo1.maven.org/maven2/");
-
                 DependencyGroup group = DependencyGroup.groupFromParameters(parameters);
 
-                repositories.stream()
+                Project project = parameters.getOriginalFile().getProject();
+
+                RepositoryLoader.loadRepositories(project).stream()
                         .map(RepositoryExplorer::new)
```

Comment:
similar thing, your `RepositoryExplorer` contains a cache, but you're not sharing them, between repositories and also between `ElementType`s despite the contents of what's in the cache being the same

Reply:
So what was happening here was that I had the caffine cache within the `RepositoryExplorer` as a static, this worked but made it hard to understand, to fix this now `repositoryExplorer` is static within the context of the completion contributor and the url is passed into the methods for getting the completions